### ISSUE TITLE
Show information about tests

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/mutation-testing-elements#865]
I want to create a "test file view" in the report. My current idea is to show the test files in the normal 

Some inspiration:

![image](https://user-images.githubusercontent.com/1828233/106324904-3a628780-627a-11eb-91b1-3507de8a531c.png)
![image](https://user-images.githubusercontent.com/1828233/106322977-f326c780-6276-11eb-89fd-60758775cfff.png)
![image](https://user-images.githubusercontent.com/1828233/106323805-4f3e1b80-6278-11eb-870a-0585e6550816.png)

If we go with a combined files table like this, then we will need to find a solution for the metrics inside the table. You might think we could simply show 2 tables back to back, 1 for the files under test and 1 for the test files. However, it is very common in JavaScript to put your unit test files _right next to your source files_, so simply splitting them out into 2 tables might not work, since then you might see 1 directory twice (once in the table with metrics for tests and once in the table for metrics for files under test). I'd personally prefer a combined view if we can manage it.

Some more thoughts:

1. If of course, the test file paths are unknown, we should add a "tests" (or something) at the bottom of the table with a view for all tests.
2. I would prefer to end up with the _actual code of the test files_ and inlining information there (as we do for mutants). I think that is the most user-friendly in the end. This would be possible for most JavaScript test runners.

**Comment by nicojs** [comment]
From #872 

> I would personally love a view on my tests and I'm personally very interested in tests that didn't kill any mutants. I'm hoping to not add more clutter than we currently have. That's why I'm thinking to add it on a separate tab, that way it is really clear that you've selected an alternative view. Prototype example:

> ![image](https://user-images.githubusercontent.com/1828233/106610821-80b03300-6567-11eb-8549-57c8ba342c8b.png)

**Comment by nicojs** [comment]
Ok, I did some work here. This is what I have now:

![test-details-demo](https://user-images.githubusercontent.com/1828233/112952311-a3f8f800-913c-11eb-90f7-4097077c47a4.gif)

1. I've reused as much as I could from the mutant view. This saved time and gets us up-and-running.
2. I've currently only worked on the test view when the source code and test locations are available. I think a simple list view for tests without locations would be fine 🤷‍♂️

@Mobrockers @hugo-vrijswijk what do you think?

**Comment by Mobrockers** [comment]
Looks good! However what is up with the location of the button? :p 

![image](https://user-images.githubusercontent.com/3595932/112988701-690aba80-9164-11eb-8f10-5ae5b7878efe.png)

**Comment by nicojs** [comment]
Yeah, for some reason `jest` is reporting the wrong columns for this tests. Nothing I can do about that, the column seems to be a "best-guess"

**Comment by nicojs** [comment]
I've actually fixed it by searching for the first _non-alfanumeric_ character.

![image](https://user-images.githubusercontent.com/1828233/113428215-b1c9aa00-93d6-11eb-9926-53e3abdae65f.png)

I've thought about it and I think this makes sense since the location of tests will almost always be best-effort and I don't see a reason why you would want the badge to appear mid-sentence. However, I don't want to create a parser, so only simple fixes like this.

In the same vein, I've implemented fuzzy logic for matching the location of tests. If an exact column number for a test doesn't exist, it will be placed on the next line.
